common/cmake/v2x.cmake

```diff
@@ -0,0 +1,221 @@
+  # VPR_TEST_PBTYPE(NAME <name>)
+  # ~~~
+  #
+  # Run the pb_type.xml file through vpr to check it is valid.
+  set(oneValueArgs NAME)
+  cmake_parse_arguments(
+    VPR_TEST_PBTYPE
+    ""
+    "${oneValueArgs}"
+    ""
+    ${ARGN}
+  )
+
+  get_target_property_required(PYTHON3 env PYTHON3)
+  get_target_property(PYTHON3_TARGET env PYTHON3_TARGET)
+
+  get_file_target(PBTYPE_TARGET_NAME "${VPR_TEST_PBTYPE_NAME}.pb_type.xml")
+  get_file_target(MODEL_TARGET_NAME "${VPR_TEST_PBTYPE_NAME}.model.xml")
+
+  add_custom_command(
+    OUTPUT "${VPR_TEST_PBTYPE_NAME}.arch.xml"
+    DEPENDS
+      ${PYTHON3_TARGET}
+      ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.pb_type.xml ${PBTYPE_TARGET_NAME}
+      ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.model.xml ${MODEL_TARGET_NAME}
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vpr_pbtype_arch_wrapper.py
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/template.arch.xml
+    COMMAND
+      ${PYTHON3} ${symbiflow-arch-defs_SOURCE_DIR}/utils/vpr_pbtype_arch_wrapper.py
+      --pb_type ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.pb_type.xml
+      --output  ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.arch.xml
+    WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils
+  )
+  add_file_target(FILE "${VPR_TEST_PBTYPE_NAME}.arch.xml" GENERATED)
+
+  add_custom_command(
+    OUTPUT "${VPR_TEST_PBTYPE_NAME}.test.eblif"
+    DEPENDS
+      ${PYTHON3_TARGET}
+      ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.pb_type.xml ${PBTYPE_TARGET_NAME}
+      ${symbiflow-arch-defs_SOURCE_DIR}/utils/vpr_pbtype_to_eblif.py
+    COMMAND
+      ${PYTHON3} ${symbiflow-arch-defs_SOURCE_DIR}/utils/vpr_pbtype_to_eblif.py
+      --pb_type ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.pb_type.xml
+      --output  ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.test.eblif
+    WORKING_DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/utils
+  )
+  add_file_target(FILE "${VPR_TEST_PBTYPE_NAME}.test.eblif" GENERATED)
+
+  xml_sort(
+    NAME ${VPR_TEST_PBTYPE_NAME}_arch_merged
+    FILE ${VPR_TEST_PBTYPE_NAME}.arch.xml
+    OUTPUT ${VPR_TEST_PBTYPE_NAME}.arch.merged.xml
+  )
+
+  get_target_property_required(VPR env VPR)
+  get_target_property(VPR_TARGET env VPR_TARGET)
+
+  set(OUT_LOCAL_REL test_${VPR_TEST_PBTYPE_NAME})
+  set(OUT_LOCAL ${CMAKE_CURRENT_BINARY_DIR}/${OUT_LOCAL_REL})
+  add_custom_command(
+    OUTPUT
+      ${OUT_LOCAL_REL}/vpr.stdout
+    DEPENDS
+      ${VPR_TEST_PBTYPE_NAME}.arch.merged.xml
+      ${VPR_TEST_PBTYPE_NAME}.test.eblif
+      ${QUIET_CMD} ${QUIET_CMD_TARGET}
+      ${VPR} ${VPR_TARGET}
+    COMMAND
+      ${CMAKE_COMMAND} -E make_directory ${OUT_LOCAL}
+    COMMAND
+      ${CMAKE_COMMAND} -E chdir ${OUT_LOCAL}
+      ${QUIET_CMD} ${VPR}
+      ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.arch.merged.xml
+      ${CMAKE_CURRENT_BINARY_DIR}/${VPR_TEST_PBTYPE_NAME}.test.eblif
+      --echo_file on
+      --pack
+      --place
+      --route
+      --disp on
```

Comment:
Probably should be headless, so don't set disp on?

Reply:
Was from my debugging. Fixed.